File: GJOH6/src/task2/service/Parser.java
package task2.service;

import task2.model.PhoneContact;
import task2.model.enums.Tag;

import java.util.Map;

/**
 * The class contains methods for parsing values for contacts
 */
public class Parser {

    /**
     * Method creates contact from HashMap values
     *
     * @param values HashMap values
     * @return PhoneContact object
     */
    public PhoneContact mapToContact(Map<String, Object> values) {
        return new PhoneContact(
                values.get("firstname").toString(),
                values.get("lastname").toString(),
                values.get("phone").toString(),
                getTagByObject(values.get("tag")));
    }

    /**
     * Method creates Tag object from Object
     *
     * @param o object
     * @return Tag enum
     */
    public Tag getTagByObject(Object o) {
        for (int i = 0; i < Tag.values().length; i++) {
            if (o!=null && o.toString().equals(Tag.values()[i].toString())) {
                return Tag.values()[i];
            }
        }
        return Tag.PRIVATE;
    }
}

File: GJOH7/service/parser.py
from model.phone_contact import PhoneContact
from model.enums.tag import Tag
from model.phone_contact import PhoneContact as Contact


class Parser:
    """
    The class contains methods for parsing values for contacts
    """
    @staticmethod
    def map_to_contact(values: dict) -> 'PhoneContact':
        """
        Method creates contact from dictionary values
        :param dictionary values
        """
        phone_numbers = []
        if 'numbers' in values:
            for phone_number_values in values['numbers']:
                phone_numbers.append({
                    'number': str(phone_number_values['number']),
                    'tag': str(phone_number_values['tag'])
                })
        return PhoneContact(
            str(values['firstname']),
            str(values['lastname']),
            phone_numbers
        )

    @staticmethod
    def get_dict_from_contact(contact: 'Contact') -> dict:
        """
        Method creates dictionary from contact
        :param dictionary values
        """
        return {
            'id': str(contact.get_id()),
            'firstname': contact.get_firstname(),
            'lastname': contact.get_lastname(),
        }

    @staticmethod
    def get_tag_by_object(o) -> 'Tag':
        for tag in Tag:
            if str(o) == tag.name:
                return tag
        return Tag.PRIVATE
